core/trino-main/src/main/java/io/trino/sql/planner/optimizations/AddExchanges.java

```diff
@@ -119,13 +123,15 @@
     private final Metadata metadata;
     private final TypeOperators typeOperators;
     private final DomainTranslator domainTranslator;
+    private final StatsCalculator statsCalculator;
 
-    public AddExchanges(Metadata metadata, TypeOperators typeOperators, TypeAnalyzer typeAnalyzer)
+    public AddExchanges(Metadata metadata, TypeOperators typeOperators, TypeAnalyzer typeAnalyzer, StatsCalculator statsCalculator)
     {
         this.metadata = metadata;
         this.typeOperators = typeOperators;
         this.domainTranslator = new DomainTranslator(metadata);
         this.typeAnalyzer = typeAnalyzer;
+        this.statsCalculator = requireNonNull(statsCalculator, "statsCalculator is null");
```

Comment:
nit: rrn other params for consistency.

Reply:
yeah, but i didn't want to add more changes to this PR... I've spent already too long with it. 
i can remove `rnn` from this one, if you prefer consistency over having the check for the new field

Reply:
no, keep it as is.